### PR TITLE
StkGHO Initialization proposal

### PR DESCRIPTION
**Pull request by The-3D**

### CODE REVIEW

File: src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.sol [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * @title StkGHO Activation
+ * @author the3d.eth
+ * - Snapshot: 'https://snapshot.org/#/aave.eth/proposal/0x4bc99a842adab6cdd8c7d5c7a787ee4c0056be554fde0d008d53b45b3e795065
+ * - Discussion: https://governance.aave.com/t/arfc-upgrade-safety-module-with-stkgho/15635
+ */
+contract StkGHO_Activation_20240118 is IProposalGenericExecutor {
+  uint256 public constant STKGHO_EMISSION_PER_SECOND = 5787037037037037; // 50 AAVE/day
+  uint256 public constant DISTRIBUTION_DURATION = 3*30*86400; // three months
+  address public constant STKGHO_PROXY = 0x1a88Df1cFe15Af22B3c4c783D4e6F7F9e0C1885d;
```
Comment (sakulstra) [inline src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.sol:22]:
```suggestion
  address public constant STKGHO_PROXY = MiscEthereum.;
```
better use from address-book via [AaveSafetyModule](https://github.com/bgd-labs/aave-address-book/blob/main/src/AaveSafetyModule.sol#L20).

(penging merge of https://github.com/bgd-labs/aave-proposals-v3/pull/171)

File: src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.sol [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  function execute() external {
+     IAggregatedStakeToken(STKGHO_PROXY).setDistributionEnd(
+      block.timestamp + DISTRIBUTION_DURATION
+    );
+    DistributionTypes.AssetConfigInput[]
+      memory enableConfigs = new DistributionTypes.AssetConfigInput[](1);
+    enableConfigs[0] = DistributionTypes.AssetConfigInput({
+      emissionPerSecond: uint128(STKGHO_EMISSION_PER_SECOND),
+      totalStaked: 0, // it's overwritten internally
+      underlyingAsset: AaveV3Ethereum.GHO_TOKEN
```
Comment (sakulstra) [inline src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.sol:33]: i'm relatively sure this is wrong - underlyingAsset i think needs to be the STK_GHO, no?

File: src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.s.sol [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    // create payloads
+    PayloadsControllerUtils.Payload[] memory payloads = new PayloadsControllerUtils.Payload[](1);
+
+    // compose actions for validation
+    IPayloadsControllerCore.ExecutionAction[]
+      memory actionsPolygon = new IPayloadsControllerCore.ExecutionAction[](1);
+    actionsPolygon[0] = GovV3Helpers.buildAction(
+      type(StkGHO_Activation_20240118).creationCode
+    );
+    payloads[0] = GovV3Helpers.buildPolygonPayload(vm, actionsPolygon);
```
Comment (sakulstra) [inline src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.s.sol:45]: should be mainnet no?

File: .gitmodules [hunk 4]
```diff
@@ -1,3 +1,6 @@
 [submodule "lib/aave-helpers"]
 	path = lib/aave-helpers
 	url = https://github.com/bgd-labs/aave-helpers
+[submodule "lib/stake-token"]
```
Comment (sakulstra) [inline .gitmodules:4]: plz remove the submodule.

the issue with submodules is that foundry is not quite good in detecting what is needed and what isn't so `forge build --force` will include everything on lib/stake-token and every dependency in that tree.
As you only need the interface plz just use it from here: https://github.com/bgd-labs/aave-address-book/blob/main/src/common/IStakeToken.sol

File: src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.sol [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import {AaveV3EthereumAssets} from 'aave-address-book/AaveV3Ethereum.sol';
+
+/**
+ * @title StkGHO Activation
+ * @author the3d.eth
+ * - Snapshot: 'https://snapshot.org/#/aave.eth/proposal/0x4bc99a842adab6cdd8c7d5c7a787ee4c0056be554fde0d008d53b45b3e795065
+ * - Discussion: https://governance.aave.com/t/arfc-upgrade-safety-module-with-stkgho/15635
+ */
+contract StkGHO_Activation_20240118 is IProposalGenericExecutor {
+  uint256 public constant STKGHO_EMISSION_PER_SECOND = 5787037037037037; // 50 AAVE/day
```
Comment (brotherlymite) [inline src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.sol:20]: This is in practise 500 AAVE / day no? or I'm I missing something

File: src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.t.sol [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+contract StkGHO_Activation_20240118_Test is ProtocolV2TestBase {
+  struct Changes {
+    address asset;
+    uint256 reserveFactor;
+  }
+
+  StkGHO_Activation_20240118 internal proposal;
+
+  function setUp() public {
+    vm.createSelectFork(vm.rpcUrl('ethereum'), 51940815);
```
Comment (brotherlymite) [inline src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.t.sol:21]:
```suggestion
    vm.createSelectFork(vm.rpcUrl('mainnet'), 19040608);
```

File: src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.t.sol [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {AaveV3EthereumAssets} from 'aave-address-book/AaveV3Ethereum.sol';
+import {AaveSafetyModule} from 'aave-address-book/AaveSafetyModule.sol';
+
+/**
+ * @dev Test for StkGHO_Activation_20240118
+ * command: make test-contract filter=StkGHO_Activation_20240118
+ */
+contract StkGHO_Activation_20240118_Test is ProtocolV2TestBase {
+  uint256 public constant STKGHO_EMISSION_PER_SECOND = 578703703703704; // 50 AAVE/day
```
Comment (foodaka) [inline src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.t.sol:22]:
```suggestion
  uint128 public constant AAVE_EMISSION_PER_SECOND = uint128(50e18) / 1 days;
```

File: src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.t.sol [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  uint256 public constant DISTRIBUTION_DURATION = 3 * 30 * 86400; // three months
+  address public constant STKGHO_PROXY = 0x1a88Df1cFe15Af22B3c4c783D4e6F7F9e0C1885d; // AaveSafetyModule.STK_GHO;
+
+  struct Changes {
```
Comment (foodaka) [inline src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.t.sol:26]: @The-3D do you plan to use?

File: src/20240118_StkGHO_Activation/StkGHOActivation.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
+---
+title: "GHO Safety Module Initialization"
+author: "Aave_BGDLabs_ACI"
```
Comment (eboadom) [inline src/20240118_StkGHO_Activation/StkGHOActivation.md:3]: Similar to other proposals we review, we are not to be included here. Usually who implements the proposal and who creates it are included (Aave Lab & ACI).

File: src/20240118_StkGHO_Activation/StkGHOActivation.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+Base emission: 50 AAVE/day
+Duration: Three months
+
+## References
+
+[TBD]
+
+## Disclaimer
+
+Aave Labs, BGD Labs and ACI receive no compensation beyond Aave protocol for the creation of this proposal. BGD Labs and ACI are both delegates within the Aave ecosystem.
```
Comment (eboadom) [inline src/20240118_StkGHO_Activation/StkGHOActivation.md:29]: No point on mentioning BGD Labs here, and we are not really delegates

File: src/20240118_StkGHO_Activation/StkGHOActivation.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+The GHO Safety Module will fortify the Aave Protocol’s resilience by adding a stablecoin asset, which is inherently less volatile than AAVE. This strategic move diversifies the Safety Module’s capacity to absorb shocks from various risk vectors in case of shortfall events.
+
+## Specification
+
+The GHO Safety module will be activated with the following parameters:
+
+Base emission: 50 AAVE/day
+Duration: Three months
+
+## References
```
Comment (eboadom) [inline src/20240118_StkGHO_Activation/StkGHOActivation.md:23]: Here links should be included. Some default ones should have been auto-generated though.
And probably good to add stkGHO proxy additionally and link to https://github.com/bgd-labs/stake-token (the implementation)

File: src/20240118_StkGHO_Activation/StkGHOActivation.md (hunk 9 above)
Comment (foodaka) [inline src/20240118_StkGHO_Activation/StkGHOActivation.md:3]:
```suggestion
author: "Aave Labs & ACI"
```

File: src/20240118_StkGHO_Activation/StkGHOActivation.md (hunk 10 above)
Comment (foodaka) [inline src/20240118_StkGHO_Activation/StkGHOActivation.md:29]:
```suggestion
Aave Labs and ACI receive no compensation beyond Aave protocol for the creation of this proposal. ACI are delegates within the Aave ecosystem.
```

File: src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.sol [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+import {IProposalGenericExecutor} from 'aave-helpers/interfaces/IProposalGenericExecutor.sol';
+import {MiscEthereum} from 'aave-address-book/MiscEthereum.sol';
+import {AaveV3Ethereum, AaveV3EthereumAssets} from 'aave-address-book/AaveV3Ethereum.sol';
+import {AaveSafetyModule} from 'aave-address-book/AaveSafetyModule.sol';
+import {IStakeToken} from './IStakeToken.sol';
+
+/**
+ * @title StkGHO Activation
+ * @author the3d.eth
```
Comment (foodaka) [inline src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.sol:12]:
```suggestion
 * @author Aave Labs & ACI
```

File: src/20240118_StkGHO_Activation/StkGHOActivation.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+[TBD]
```
[the remark this answers is not in the document]
Comment (foodaka) [reply src/20240118_StkGHO_Activation/StkGHOActivation.md:25]: @The-3D wanna add the simulated payload address?

File: src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.t.sol [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+
+/**
+ * @dev Test for StkGHO_Activation_20240118
+ * command: make test-contract filter=StkGHO_Activation_20240118
+ */
+contract StkGHO_Activation_20240118_Test is ProtocolV2TestBase {
+  uint128 public constant AAVE_EMISSION_PER_SECOND = uint128(50e18) / 1 days; // 50 AAVE per day
+  uint256 public constant DISTRIBUTION_DURATION = 90 days; // 3 months
+
+  struct Changes {
```
Comment (foodaka) [inline src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.t.sol:25]:
```suggestion

```

File: src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.t.sol [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+    address asset;
+    uint256 reserveFactor;
```
Comment (foodaka) [inline src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.t.sol:27]:
```suggestion

```

File: src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.t.sol [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+  }
```
Comment (foodaka) [inline src/20240118_StkGHO_Activation/StkGHO_Activation_20240118.t.sol]:
```suggestion

```

**Comment by miguelmtzinf** [comment]
AIP rebuilt with autogenerate tool and suggestions amended.

**Review by eboadom — APPROVED** [review]

**Review by brotherlymite — APPROVED** [review]

**Pull request merged**